zio-cli/shared/src/main/scala/zio/cli/PrimType.scala

```diff
@@ -36,20 +37,93 @@ object PrimType {
   }
 
   final case class Path(pathType: PathType, exists: Boolean) extends PrimType[JPath]
-  case object Text                                           extends PrimType[String]
-  case object Decimal                                        extends PrimType[BigDecimal]
-  case object Integer                                        extends PrimType[BigInt]
-  case object Instant                                        extends PrimType[JInstant]
-  case object LocalDate                                      extends PrimType[JLocalDate]
-  case object LocalDateTime                                  extends PrimType[JLocalDateTime]
-  case object LocalTime                                      extends PrimType[JLocalTime]
-  case object MonthDay                                       extends PrimType[JMonthDay]
-  case object OffsetDateTime                                 extends PrimType[JOffsetDateTime]
-  case object OffsetTime                                     extends PrimType[JOffsetTime]
-  case object Period                                         extends PrimType[JPeriod]
-  case object Year                                           extends PrimType[JYear]
-  case object YearMonth                                      extends PrimType[JYearMonth]
-  case object ZonedDateTime                                  extends PrimType[JZonedDateTime]
-  case object ZoneId                                         extends PrimType[JZoneId]
-  case object ZoneOffset                                     extends PrimType[JZoneOffset]
+  {
+    import PathType._
+    def validate(value: String): IO[String,JPath] = {
```

Comment:
This method is performing side effects and needs to be made lazy, see IO.effectSuspendTotal